storage/bootstrap/bootstrapper/commitlog/source_prop_test.go

```diff
@@ -0,0 +1,235 @@
+package commitlog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+	"github.com/m3db/m3db/context"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/ts"
+	xtime "github.com/m3db/m3x/time"
+)
+
+func TestCommitLogSourcePropCorrectlyBootstrapsFromCommitlog(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.Rng.Seed(123456789)
+	parameters.MinSuccessfulTests = 100
+	props := gopter.NewProperties(parameters)
+
+	props.Property("Commitlog bootstrapping properly bootstraps the entire commitlog", prop.ForAll(
+		func(input propTestInput) (bool, error) {
+			// Make sure we have a new directory for each test run
+			dir, err := ioutil.TempDir("", "foo")
```

Comment:
do you need to cleanup this dir once you're done?

Reply:
Won't this cause issues because it generates a random directory id internally. i.e. gopter shrinking/repro should fail no?

Reply:
1) Regarding the temp dir, do I? I looked through existing test cases and they don't look like they cleanup old directories. I figured the O.S would just reclaim it eventually

2) Does the random directory ID matter? This test writes the commitlog files and reads them back all in one go. If gopter is trying to repro, it will just be producing new data into new folders which should be fine. Maybe I'm misunderstanding something?

Reply:
Yep you're right. It's an issue if you're using a SUT, not with this flavor of gopter.